src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -2514,18 +2536,144 @@ public void Dispose()
                 return;
             }
 
-            this.DisposingEvent.SafeInvoke(this, EventArgs.Empty);
-            commandRuntime = null;
-            currentObjectInPipeline = null;
-            _input.Clear();
-            // _scriptBlock = null;
-            // _localsTuple = null;
-            // _functionContext = null;
+            try
+            {
+                if (_scriptBlock.HasCleanupBlock)
+                {
+                    InvokeCleanupBlock();
```

Comment:
Yeah, I hear you. Given what the feature is trying to accomplish it seemed the simplest way to approach it, but it probably would be preferable to avoid things potentially crashing the whole application.